data/fields/fuel/fuel_multi.json

```diff
@@ -8,6 +8,7 @@
             "GTL_diesel": "GTL Diesel",
             "HGV_diesel": "Truck Diesel",
             "biodiesel": "Biodiesel",
+            "diesel:class2": "Cold-weather Diesel",
```

Comment:
I don’t think this should be added to the global `fuel:*` field, because “class 2” only means winter diesel or arctic diesel in some countries. The key `diesel:class2` would be very confusing in other countries, for example the U.S., where “Diesel #<!-- -->2” is ordinary diesel, but there is winter blend diesel and winterized diesel, which would have to be tagged differently because it doesn’t meet the EU’s EN590 standard as described on the wiki page.

As we add fuel types specific to other countries, we should split out new field JSON files for those countries. That will allow us to tie the string more closely to the fuel’s name in each country, so that users don’t have to be well-versed in the international oil trade to tag what they see on the gas station’s signage.

Reply:
I admit I am not familiar with all of the winterised diesel standards worldwide, however this seems like a tag applicability rather than a schema issue? Using the same argument, one could think that biodiesel and "HGV diesel" may mean different, equally confusing, things to mappers in different countries. This will not be resolved until a single standard is implemented - which is unrealistic. From memory, there was similar confusion about octane tagging and how this is assessed in the US.
The wiki definition seems quite broad, and can be applied to various winterised diesels, however the tag only needs to be specific enough for a mapper seeing "Winter diesel" to be able to tag that. "class2" is a more generic term, looking at tag usage worldwide, than may seem.

Reply:
The difference with `biodiesel` and `hgv_diesel` is that we aren’t taking a more general concept and applying a key that says something highly specific to a particular country or region’s regulations. The analogy with octane numbers supports my point: U.S. mappers are tagging the signposted octane numbers as is, without converting them to European octane numbers (which would be impossible).

[The wiki table](https://wiki.openstreetmap.org/wiki/Special:PermanentLink/2429851#How_to_map) currently defines `fuel:diesel:class2` as:

> [Diesel for colder weather climates](https://en.wikipedia.org/wiki/Winter_diesel_fuel#Arctic_diesel) (with a threshold filtration temperature of −32 °C), on the EU EN590 scale. Can be branded as "winter diesel", "Arctic diesel", "diesel −32 °C", etc., depending on the location.

It goes on to describe a `fuel:diesel:class0` for certain winter diesels. You may consider the definition to be broad, but it’s tied to an EU standard. In regions that make different distinctions, no one would be able to map these keys back to a real-world fuel type for end users.

`fuel:diesel:class2` is only being used in Europe, with the exception of [one gas station in the U.S.](https://www.openstreetmap.org/way/824731205) where the mapper probably didn’t fully comprehend the definition on the wiki.

To be clear, I see no problem with adding `fuel:diesel:class2` to fuel_multi-RU.json, but adding it to fuel_multi.json would make it more difficult for us to promote common-sense tagging outside Europe. I think you should be able to add a `strings` object to fuel_multi-RU.json that contains just the string for `diesel:class2`, and schema-builder should continue to cross-reference fuel_multi.json for all the other strings.

Reply:
See your point (although I think "Truck diesel" for HGV diesel is even worse of a generalisation. To me, the wiki reads like it would apply to any diesel with a filtration temp of -32.
I've changed the commit to only touch the RU multi box (assume it is specific to Russia, they use the same standard)

Reply:
Actually, thinking about it, how about adding this with a more specific label eg "EU winter diesel", given there is no EU JSON that I see (Russia seems to be the only one)

Reply:
> To me, the wiki reads like it would apply to any diesel with a filtration temp of -32.

That’s part of the conundrum. The U.S. and Canada [define](https://en.wikipedia.org/wiki/Winter_diesel_fuel#United_States) winter blend diesel in terms of relative temperature, not on an absolute temperature scale

>  I've changed the commit to only touch the RU multi box (assume it is specific to Russia, they use the same standard)

You’ll want to also add the `strings` object to this file as I mentioned above. Otherwise, there would be no opportunity to translate `class2` to Russian in Transifex.

> Actually, thinking about it, how about adding this with a more specific label eg "EU winter diesel", given there is no EU JSON that I see (Russia seems to be the only one)

We could create a field for the whole EU just like there is for Russia. The `locationSet` property is evaluated by [country-coder](https://github.com/ideditor/country-coder/), which supports `EU` as a location.

Reply:
Getting out of my depth here - is copying the `strings` section from global Fuel Multi enough, with class2 added?

Given how I am not familiar with the schema to be able to tell, what other dependencies there are for creating a location-specific multibox